core/root/config/config.go

```diff
@@ -88,6 +88,14 @@ func ConfigFromBytes(data []byte) (*Config, error) {
 	}
 
 	for i, np := range nodePools {
+		if err := np.Experimental.Taints.Valid(); err != nil {
+			return nil, fmt.Errorf("invalid taints for node pool at index %d: %v", i, err)
+		}
+
+		// Controllers must be aware of node pools' taints so they know which tolerations
+		// to add to pods that are supposed to run on every node (ex: kube-proxy)
+		cpCluster.ComputedNodeSettings.NodeTaints = append(cpCluster.ComputedNodeSettings.NodeTaints, np.Experimental.Taints...)
```

Comment:
From what I've been testing, running kube-proxy as a DaemonSet as of now may indeed not be the best choice after all. If you guys think it's best to leave this to the future, I'm okay with that!